File: mpp-proiect-repository-cristinahendre/lab8(bun)/CSharp/Lab5/Service/IService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Cerinta1.Domain;

namespace Cerinta1.Service
{
    public interface IService
    {
        Voluntar login(String e, String p, IObserver obs);
        void logout(Voluntar user, IObserver client);

        CazCaritabil findCaz(int id);

        Voluntar getVoluntarDupaDate(String e, String p);
        void saveDonatie(Donatie don, IObserver client);
        Donator getDonatorDupaDate(String n, String p, String ad, long tel);
        void saveDonator(String n, String p, String ad, long tel);

        IEnumerable<CazCaritabil> getCazuri();

        IEnumerable<Donator> getDonatori();

        IEnumerable<Donator> getDonatorDupaNume(String n);
    }
}

File: mpp-proiect-repository-cristinahendre/lab8(bun)/Java/lab6/modul/ServiceModule/src/main/java/Service/IService.java
package Service;


import Domain.CazCaritabil;
import Domain.Donatie;
import Domain.Donator;
import Domain.Voluntar;

public interface IService {
    Voluntar login(String e, String p, IObserver client) throws  ServiceException;

    void logout(Voluntar user, IObserver client) throws ServiceException;
    Voluntar getVoluntarDupaDate(String e, String p);
    void saveDonatie(Donatie don,  IObserver client);
    Donator getDonatorDupaDate(String n, String p, String ad, long tel);
    void saveDonator(String n, String p, String ad, long tel);

    Iterable<CazCaritabil> getCazuri();

    Iterable<Donator> getDonatori();

    Iterable<Donator> getDonatorDupaNume(String n);


}
